prow/cmd/checkconfig/main.go

```diff
@@ -365,6 +368,12 @@ func validate(o options) error {
 		}
 	}
 
+	if o.warningEnabled(validateSupplementalProwConfigOrgRepoHirarchy) {
+		for _, supplementalProwConfigDir := range o.supplementalProwConfigDirs.Strings() {
+			errs = append(errs, validateAdditionalProwConfigIsInOrgRepoDirectoryStructure(filepath.Dir(supplementalProwConfigDir), os.DirFS("/")))
```

Comment:
One weirdness now is that checkconfig needs a flag set for each org, which is error-prone. Alternatively we could probably add a new flag on top of `supplementalProwConfigDirs` like `supplementalProwConfigsBaseDir`?